core/trino-main/src/main/java/io/trino/sql/analyzer/RelationType.java

```diff
@@ -104,7 +103,7 @@ public int getVisibleFieldCount()
      */
     public Collection<Field> getAllFields()
     {
-        return ImmutableSet.copyOf(allFields);
```

Comment:
Do you know why it was implemented this way? It looks like optimization of something.

Reply:
It was to make containment checks cheaper, but it's possible that no longer matters.

Reply:
At this point it is used only for iteration.